### PR TITLE
refactor(product): move core/product into the product module

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

Issue Number: https://github.com/graycoreio/daffodil/issues/152

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/product/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    "type": "git",
+    "url": "https://github.com/graycoreio/daffodil"
+  },
+  "author": "Graycore LLC",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/graycoreio/daffodil/issues"
+  },
+  "scripts": {
+    "build": "ng build product",
```
Comment (damienwebdev) [inline libs/product/package.json]: This should be `ng build --prod`.

File: libs/product/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "lint": "ng lint product",
+    "test": "ng test product --watch=false --browsers=ChromeHeadless"
```
Comment (damienwebdev) [inline libs/product/package.json]: Let's leave this out temporarily until we have something to test.

File: libs/product/package.json (hunk 1 above)
Comment (lderrickable) [reply libs/product/package.json]: done

File: libs/product/package.json (hunk 2 above)
Comment (lderrickable) [reply libs/product/package.json]: removed

File: libs/product/package.json [hunk 3]
```diff
@@ -12,9 +12,8 @@
     "url": "https://github.com/graycoreio/daffodil/issues"
   },
   "scripts": {
-    "build": "ng build product",
-    "lint": "ng lint product",
-    "test": "ng test product --watch=false --browsers=ChromeHeadless"
+    "build": "ng build --prod",
```
Comment (damienwebdev) [inline libs/product/package.json]: `ng build product --prod`

Comment (lderrickable) [reply libs/product/package.json]: Yeah, I thought that was odd, but I thought I'd take your word for it

Comment (lderrickable) [reply libs/product/package.json]: done

**Comment by damienwebdev** [comment]
Needs squash.

**Pull request merged**